### PR TITLE
eeprom_metadata.py - auto .hex to .bin

### PR DESCRIPTION
**Pull request by nullstalgia**

### CODE REVIEW

**Comment by mihaigalos** [comment]
Awesome, @nullstalgia. Please edit the first comment and provide some context. I'm adding my review now.

File: utils/eeprom_metadata.py [hunk 1]
```diff
@@ -131,15 +147,34 @@ def getArguments():
...
+    dont_auto_convert = args['dont_convert']
+    return binary_file, application_name, output_file, print_header_result, dont_auto_convert
+
+def check_if_binary(check):
+    file_name, file_ext = splitext(check)
+    if file_ext == ".bin":
+        return True, check
+    elif file_ext == ".hex":
+        if not dont_auto_convert:
+            # check=True is used to raise an exception if the command does not run successfully (doesn't exist, bad file, etc)
```
Comment (mihaigalos) [inline utils/eeprom_metadata.py]: you don't need this comment here. If you still think you need to tell the user your intent, please refactor this to a new free function.

File: utils/eeprom_metadata.py [hunk 2]
```diff
@@ -131,15 +147,34 @@ def getArguments():
...
+            subrun(['avr-objcopy', '-I', 'ihex', binary_file, '-O', 'binary', file_name+'.bin'], check=True)
+            # Return the new file's path
```
Comment (mihaigalos) [inline utils/eeprom_metadata.py]: Superfluous, remove.

**Review by mihaigalos — CHANGES_REQUESTED** [review]

File: utils/eeprom_metadata.py (hunk 1 above)
Comment (nullstalgia) [reply utils/eeprom_metadata.py]: Apologies, I tend to over-comment :)

File: utils/eeprom_metadata.py (hunk 2 above)
Comment (nullstalgia) [reply utils/eeprom_metadata.py]: If you are referring to the comment on line 161, that has been removed in the newest commit as well.

**Review by mihaigalos — APPROVED** [review]
awesome, thank you!

**Pull request merged**